external-crates/move/crates/move-vm-runtime/src/cache/move_cache.rs

```diff
@@ -81,6 +84,22 @@ impl MoveCache {
         &self.package_cache
     }
 
+    // -------------------------------------------
+    // Loading and Resolving Packages
+    // -------------------------------------------
+
+    /// Resolve a package, loading it if necessary. This will use the provided `ModuleResolver` to
+    /// fetch the package if it is not already cached.
+    /// If the package is not found, `Ok(None)` is returned.
+    /// If there is an error loading or verifying the package, an error is returned.
```

Comment:
This type is miserable. Can we do something more-informative than "None means it was not in the ModuleResolver"? Even just a little enum to let the code self-document.